Chapter3/src/wci/frontend/java/JavaScanner.java

```diff
@@ -24,7 +25,7 @@ public JavaScanner(Source source) {
 	@Override
 	protected Token extractToken() throws Exception {
 		skipWhiteSpace();
-		Token token;
+		Token token = null;
```

Comment:
This is only temporary. I got build error if not set to null as token is not guaranteed to get initialized currently.